apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -179,31 +200,34 @@ export class SendMessageInApp extends SendMessageBase {
     const bridgeOutputs = command.bridgeData?.outputs as InAppOutput;
     const inAppMessage = inAppMessageFromBridgeOutputs(bridgeOutputs);
 
+    const channelData: Partial<Pick<MessageEntity, 'content' | 'subject' | 'avatar' | 'payload' | 'cta' | 'tags'>> = {
+      content: (this.storeContent() ? inAppMessage.content || content : null) as string,
+      cta: bridgeOutputs ? inAppMessage.cta : step.template.cta,
+      subject: inAppMessage.subject,
+      avatar: inAppMessage.avatar,
+      payload: messagePayload,
+      tags: command.tags,
+    };
```

Comment:
Constructing the channel data for shared use in both message creation and update logic.